src/main/io/statusindicator.c

```diff
@@ -60,20 +76,32 @@ void warningLedBeeper(bool on)
 void warningLedPulse(void)
 {
 	// FIXME this is not really pulse now - original code pulsed the light once a second (on short, off long), this code is a result of rebasing onto commit beac0a35cec690f33a2ef5d13f41f3e3a9d8e57a
-	nextBlinkMask = 0b10000000;
+	nextBlinkMask = 0b1000001;
 }
 
 void warningLedSetBlinkMask(uint32_t newBlinkMask)
 {
 	nextBlinkMask = newBlinkMask;
 }
 
+void warningLedCode(uint8_t code)
+{
+    nextBlinkMask = 0b100000;
+ 	int bitsRemaining = 32 - 6;  // must be an even number
+    for (int i = 0; i < code && bitsRemaining; i++) {
```

Comment:
Yes, I understand that.
After working with micros for some time, the bit handling idioms look quite natural ..
maybe using macros may make it quite easy to read ... 
 ```
#define NBITS(n) ((1 << (n)) - 1)
(NBITS(code * 2) & 0x55555555) | (1 << (code * 2 + trailZeros)) 
```

In loop code, it will be better to test ` nextBlinkMask & (3 << 29) ` for remaining space instead of bitsRemaining - you want to stop before shifting end marker out.